docs/proposals/instrumentation-of-services.md

```diff
@@ -0,0 +1,70 @@
+# Best practices for instrumenting infrastructure components in OpenShift
+
+All OpenShift components MUST expose the following endpoints on either their public HTTPS port, or a public HTTP port if none exists
+
+* `/healthz` - returns 200 and the text `ok` if the instance is alive
+* `/metrics` - returns a reasonable set of metrics in the Prometheus format for capturing the health and performance of the server in question
```

Comment:
For internal cluster management components which also happen to have public routes, it seems we'd want to keep `/metrics` routable only internally. Perhaps `/metrics` would be a "SHOULD" rather than a "MUST" with regards to public endpoints?